### DISCUSSION TITLE
[email] Error: You don't have permission to access this

### DISCUSSION THREAD

**Discussion by Ivann-2612** [directus/directus#13448] [Archive]
### Describe the Bug

Whwn i click on forgot passord button and put the an email after that i have this message in termianl.I dont know what is the problem of this?

### To Reproduce

I set up env file and i have this problem

### Errors Shown

[email] Error: You don't have permission to access this

### What version of Directus are you using?

last version

### What version of Node.js are you using?

v14.15.4

### What database are you using?

SQLite3

### What browser are you using?

Chrome

### How are you deploying Directus?

Locally

**Comment by rijkvanzanten** [comment]
How did you setup the env file? How are you running Directus? What does your email configuration look like? I'm not able to reproduce this on a fresh installation nor on Cloud, so I believe this is a configuration problem

**Comment by Ivann-2612** [comment]
My env setup is:

EMAIL SMTP HOST=‘hosting provider’
EMAIL SMTP PORT=587
EMAIL SMTP USER=‘myemail’
EMAIL SMTP PASSWORD=‘password’
EMAIL SMTP POOL=false
EMAIL SMTP SECURE=false

Google chrome :
Less secure app access is enabled

I am running Directus locally in my localhost:8055

**Comment by Ivann-2612** [comment]
I think something is wrong with the env file, that I didn't set
something up properly but don’t know what

**Comment by Ivann-2612** [comment]
This is a whole env configuration:


EMAIL_TRANSPORT='"smtp"
EMAIL_SMTP HOST='smtp. gmail.com'
EMAIL_SMTP =***@***.***
EMAIL_SMTP PASSWORD=‘********’
EMAIL_SMTP SECURE= false
EMAIL_SMTP PORT= 587
EMAIL_SMTP POOL= true
EMAIL_SMTP NEW LINE-"unix
EMAIL_SMTP PATH-''/usr/sbin/sendmail"

An error in the terminal console:
https://user-images.githubusercontent.com/60244706/169643324-3090c4e9-bb63-4d2b-b33d-fa5ea7a5dfa9.jpeg

**Reply by Powma** [reply]
Maybe the system don't have access in your file .env. Are you on Linux ?

**Comment by Ivann-2612** [comment]
No, my OS is Windows 10

**Comment by Ivann-2612** [comment]
![Screenshot 2022-05-23 095021](https://user-images.githubusercontent.com/60244706/169770409-31054bd9-0084-4743-8e50-167bc75c7efc.png)
This is a screenshot from  the Postman, do you know using this what the point is?

and this message in the console
![Screenshot 2022-05-23 101441](https://user-images.githubusercontent.com/60244706/169774807-42b695c8-412d-4b1a-b369-671b6ff5a30d.png)

**Reply by azrikahar** [reply]
The `Greeting never received` error seems to be similar to this: nodemailer/nodemailer#441

With that info and based on the env values you shared here: https://github.com/directus/directus/discussions/13448#discussioncomment-2795110, you probably need to set `EMAIL_SMTP_PORT` to `465` and try ensuring `EMAIL_SMTP_SECURE` is set to `true`. 

Note: make sure you are setting them correctly, as it seems like you pasted `EMAIL_SMTP SECURE` with a missing underscore instead of `EMAIL_SMTP_SECURE`, but I assume that is just a typo in this discussion, not what you actually set in your env.

**Reply by Ivann-2612** [reply]
I tried as you said and always brings me back to this console error.
![Screenshot 2022-05-23 103923](https://user-images.githubusercontent.com/60244706/169779750-e7461433-7dcb-4026-9cab-aae9a546a012.png)

**Reply by azrikahar** [reply]
did you send a payload with "email" in it? Based on your postman screenshot shared above, it seems like you weren't sending it.

Example payload:

```json
{
    "email": "admin@example.com"
}
```

(Of course this is assuming you are getting the permission error over here, which meant the user object is null, which subsequently meant the email was not specified or the email specified does not exist)

https://github.com/directus/directus/blob/a7534c1211e7c0dfdca98ac4caced0a89561c43e/api/src/services/users.ts#L341-L344

**Reply by Ivann-2612** [reply]
You don't have permission is an error when I try to run this env file setting in Directus forgot
password option on my email

**Reply by azrikahar** [reply]
The code snippet I shared is indeed within the passwordResetRequest function, which is called by the `/auth/password/request` POST endpoint. And since your error logged has the `[email]` prefix, my understanding is that it is more likely for the error to happen around here.

https://github.com/directus/directus/blob/a7534c1211e7c0dfdca98ac4caced0a89561c43e/api/src/controllers/auth.ts#L140-L169

Just to confirm based on the snippet I shared earlier, is the user of the email you requested _confirmed_ to have "active" status?

**Reply by Ivann-2612** [reply]
Thanks for your help so far, I'll continue looking into the issue and let you know if I have some further questions.

**Reply by Ivann-2612** [reply]
What do you think ofthis LOG_LEVEL=trace message in the console?

  select `ip_access` from `directus_roles` where `id` is null limit ? [1]
12:04:38 ­ƒöì [0.503ms] select `status`, `password` from `directus_users` where `email` = ? limit ? [ivan.nikolicamp@gmail.com, 1] 

This is the message when I wanted to send to my email forgot password option

**Reply by azrikahar** [reply]
Thanks for the additional info! Definitely seems like query right before the IF statement I shared earlier, which is (line 339 query):

https://github.com/directus/directus/blob/6c50c5e55e28b36a4059bad9ef448b3b10d68bd3/api/src/services/users.ts#L339-L344

Were you able to check the user's status? Based on the logic, it will throw forbidden exception if the user's status is not "active", which seems to be the only possibility here 🤔

**Comment by developer-appdam** [comment]
Same issue here...

**Comment by mqwerty** [comment]
I get this error if there is no user with that email address.
